packages/components/src/drop-zone/provider.js

```diff
@@ -233,13 +233,13 @@ class DropZoneProvider extends Component {
 	}
 
 	render() {
-		const { children } = this.props;
-		return children;
+		return (
+			<Provider value={ this.dropZoneCallbacks }>
```

Comment:
Although this is the same as before: how about making `DropZone` a `PureComponent`? That way, we prevent doing re-renders every time we `setState` in this component. We already take care of telling dropZones to render individually ([on reset](https://github.com/WordPress/gutenberg/pull/11100/files#diff-af960f1a0848b34c0c03399a679e8c4dR119), and [if the dropZone needs updating](https://github.com/WordPress/gutenberg/pull/11100/files#diff-af960f1a0848b34c0c03399a679e8c4dR184).

Reply:
mmm, I'm not sure I follow. Let me clarify what I'm saying:

* we render the `DropZone` by explicitly calling its `setState` method from the `DropZoneProvider` in the places I mentioned.
* we also render the `DropZone` by the normal React mechanism: everytime the `DropZoneProvider` is rendered its children also are. This is the case I think we may want to prevent by converting `DropZone` to a `PureComponent`.

Does it make more sense now?

Reply:
Not really 🤷‍♂️ How do you transform it to a pure component while keeping `setState`?

Reply:
Oh you mean `pure` by wrapping it in our `pure` HoC :). In theory, this shouldn't be necessary. The question could be, why do we not wrap all of our components with it?

The response is that it should probably be optimized on a upper level. and this could hurt more than it helps.

Reply:
`pure` works for both functional and class component. `PureComponent` is a base class only.